### PR TITLE
Sscs 10286 add handling error state

### PR DESCRIPTION
**Pull request by Valantinas**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/SSCS-10286

### Change description ###
New hearing state ERROR_HANDLING has been added.
This PR also contains unmerged (at the time of writing) changes from another PR 10245(added epimsId) which is a dependency for the implementation of the ticket above.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by MohammedHMCTS — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/SscsCaseData.java [hunk 1]
```diff
@@ -589,6 +585,22 @@ public WorkAllocationFields getWorkAllocationFields() {
...
+    }
+
+    @JsonIgnore
+    public JointParty getJointParty() {
+        if (isNull(jointParty)) {
+            this.jointParty = new JointParty();
+        }
+        return jointParty;
+    }
+
```
Comment (nickwest1) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/SscsCaseData.java:603]: Why are we using isNull in one place and == null in another?  We should have consistency.

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/SscsCaseData.java [hunk 2]
```diff
@@ -613,6 +625,14 @@ public Optional<LocalDateTime> getDateTimeSentToGaps() {
...
     }
 
+    @JsonIgnore
+    public SscsIndustrialInjuriesData getSscsIndustrialInjuriesData() {
+        if (sscsIndustrialInjuriesData == null) {
+            this.sscsIndustrialInjuriesData = new SscsIndustrialInjuriesData();
+        }
+        return sscsIndustrialInjuriesData;
+    }
+
```
Comment (nickwest1) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/SscsCaseData.java:635]: Same here RE isNull comment above

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/YesNo.java [hunk 3]
```diff
@@ -32,15 +33,15 @@ public static boolean isYes(YesNo yesNo) {
...
     }
 
     public static boolean isNoOrNull(YesNo yesNo) {
         return isNull(yesNo) || NO.equals(yesNo);
     }
 
     public static boolean isNoOrNull(String yesNo) {
-        return isNull(yesNo) || NO.getValue().equals(yesNo);
+        return isNull(yesNo) || NO.getValue().equalsIgnoreCase(yesNo);
     }
```
Comment (nickwest1) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/YesNo.java:45]: Why are we not simply doing Boolean.parseBoolean(yesNo) ?
The YesNo class just seems to add code for the sake of it.
NB: For the sake of time, don't go changing this, but just needed to raise this question somewhere :)

**Review by nickwest1 — CHANGES_REQUESTED** [review]
Be handy to have an answer RE the YesNo class, but otherwise just deal with the other two comments please

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/YesNo.java (hunk 3 above)
Comment (bensonlievoco) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/YesNo.java:45]: Boolean.parseBoolean(yesNo) , if the data is "yes" will return false

Comment (nickwest1) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/YesNo.java:45]: My bad, I meant this:
BooleanUtils.toBoolean(String)

It's an apache commons class.

**Pull request closed without merging**